app/streaming/CartCountContext.tsx

```diff
@@ -0,0 +1,36 @@
+'use client';
+
+import React, { useState } from 'react';
+
+const CartCountContext = React.createContext<
+  [number, React.Dispatch<React.SetStateAction<null | number>>] | undefined
+>(undefined);
+
+export function CartCountProvider({
+  children,
+  initialCartCount,
+}: {
+  children: React.ReactNode;
+  initialCartCount: number;
+}) {
+  const [optimisticCartCount, setOptimisticCartCount] = useState<null | number>(
+    null,
```

Comment:
Should this be set to `0` instead of `null`?

I see we're [defaulting to `0` if we cannot read the cookie value](https://github.com/vercel/app-playground/pull/60/files#diff-441a26f1107abe3ce7f4027d4031e34c77e4fce7319dadbf18ecdf65b2593499R10).

I guess it also depends if we want to show `0` or show no red cart count bubble at all (which is where I could see `null` working here).

Reply:
Discussed internally. This is the recommended pattern.